apps/sms/views/conversation/js/conversation.js

```diff
@@ -2351,7 +2215,7 @@ var ConversationView = {
         this.shouldChangePanelNextEvent = false;
         Navigation.toPanel('thread-list');
         if (ActivityClient.hasPendingRequest()) {
-          setTimeout(this.close.bind(this), this.LEAVE_ACTIVITY_DELAY);
+          setTimeout(this.backOrClose.bind(this), this.LEAVE_ACTIVITY_DELAY);
```

Comment:
I think this case is not really readable because of the name change; maybe a comment saying "backOrClose is closing when in an activity" could help?